src/components/List/List.js

```diff
@@ -35,13 +33,22 @@ const List = ({ token, setToken }) => {
           <button data-testid="clearTokenButton" onClick={clearToken}>
             Return to welcome screen
           </button>
+
           <div className="grocery-list">
-            <ul>
-              {groceryItems &&
-                groceryItems.map((item) => (
+            {!loading && !error && groceryItems && groceryItems.length ? (
```

Comment:
Not necessary to change but the multiple checks in this line might be hard to debug down the line. Could move that into a `showList` function.